### ISSUE TITLE
copy.java (copyFileToPod)

### ISSUE THREAD

**Issue by sayalisaitawdekar** [kubernetes-client/java#853]
I am using copyFileToPod function, but its not working, what is wrong 
my sourcePath is : Path srcpath = Paths.get("hello.txt"); hello.txt is in my project structure.
my dest path is : Path destpath = Paths.get("/usr/share/logstash/logstashvolume"); 
/usr/share/logstash/logstashvolume: this is the folder structure of the pod in which i have to put my file. nothing is encoded. Is anything wrong?

**Comment by elanv** [comment]
Could you share a log of the problem?
And what do you mean about "nothing is encoded."?

**Comment by brendandburns** [comment]
Copy relies on the `tar` executable being present in your Pod. Is it present?

**Comment by rofl** [comment]
Today I had the same problem, seems to me a "-" is missing in the implementation:
https://github.com/kubernetes-client/java/blob/master/util/src/main/java/io/kubernetes/client/Copy.java#L183

Instead of
new String[] {"sh", "-c", "base64 -d | tar -xmf - -C " + parentPath}

it should be
new String[] {"sh", "-c", "base64 -d - | tar -xmf - -C " + parentPath}

StdIn couldn't be read and nothing happened.

Perhaps it's just the base image of the pod, a different base64 maybe. But debian:buster should be pretty standard.

**Comment by elanv** [comment]
I have tried with debian:buster and there was no problems.
And base64 in debian:buster shows help like this:
```
# base64 --help
Usage: base64 [OPTION]... [FILE]
Base64 encode or decode FILE, or standard input, to standard output.

With no FILE, or when FILE is -, read standard input.
```

So, it seems missing '-' is not the cause of the problem at least about debain:buster image.
I agree to you that different base64 or tar binary cause the problem.
Could you share more information of your docker image you have tested?

**Comment by rofl** [comment]
@elanv 
You are right. I am sorry. Yesterday I just tried too hard to get this done, and missed the actual point... The '-' is not needed. My mistake.

After some sleep, I think I found my problems.
First the `destPath `will be turned into path to parent folder:
`String parentPath = destPath.getParent() != null ? destPath.getParent().toString() : ".";`

With Windows (company policy...) the seperator is not usable on my linux pod. So I will add this to mitigate:
`parentPath = parentPath.replace("\\", "/");`

And I need to put the Filename into `destPath`. The implementation will always use "parent" to resolve the destination folder, and is using `destPath.getFilename()` to create an `TarArchiveEntry`:

`copyFileToPod(namespace, pod, container, Paths.get("/local/path/filename.txt"), Paths.get(				"/destination/on/pod/filename.txt"));`

This could be @sayalisaitawdekar problem?
